bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/ICalendarHandlerFactory.java

```diff
@@ -67,6 +84,61 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         if (!supportsThingType(thingTypeUID)) {
             return null;
         }
-        return new ICalendarHandler(thing, sharedHttpClient, eventPublisher);
+        if (thingTypeUID.equals(THING_TYPE_CALENDAR)) {
+            if (thing instanceof Bridge) {
+                return new ICalendarHandler((Bridge) thing, sharedHttpClient, eventPublisher);
+            } else {
+                // Migration needs to be done asynchronously. Using thread pool for common things.
+                ThingUID uidToCopy = thing.getUID();
+                ExecutorService threadPool = ThreadPoolManager.getPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+                threadPool.execute(new ReregisterThingRunnable(uidToCopy));
+                return null;
+            }
+        } else if (thingTypeUID.equals(THING_TYPE_FILTERED_EVENTS)) {
+            return new EventFilterHandler(thing);
+        }
+        return null;
+    }
+
+    /**
+     * This Runnable "upgrades" the thing "calendar" to the bridge of same type. To make sure the thing isn't in use, a
+     * sleep blocks before doing actual work.
+     *
+     * @author Michael Wodniok - Initial contribution
```

Comment:
The author tag should not be used on inner classes.

Reply:
Hm. Yes this was strange, i've added it as a maven plugin complained about the missing `@author`. However it now does not anymore. I'll remove.